x/shield/handler.go

```diff
@@ -178,63 +161,59 @@ func handleMsgResumePool(ctx sdk.Context, msg types.MsgResumePool, k Keeper) (*s
 	return &sdk.Result{Events: ctx.EventManager().Events()}, nil
 }
 
-func handleMsgPurchaseShield(ctx sdk.Context, msg types.MsgPurchaseShield, k Keeper) (*sdk.Result, error) {
-	_, err := k.PurchaseShieldForP(ctx, msg.PoolID, msg.Shield, msg.Description, msg.From)
-	if err != nil {
+func handleMsgWithdrawForeignRewards() (*sdk.Result, error) {
+	return &sdk.Result{}, types.ErrOperationNotSupported
+}
+
+func handleMsgDepositCollateral(ctx sdk.Context, msg types.MsgDepositCollateral, k Keeper) (*sdk.Result, error) {
+	if msg.Collateral.Denom != k.BondDenom(ctx) {
+		return nil, types.ErrCollateralBadDenom
+	}
+
+	if err := k.DepositCollateral(ctx, msg.From, msg.Collateral.Amount); err != nil {
 		return nil, err
 	}
 
 	ctx.EventManager().EmitEvents(sdk.Events{
 		sdk.NewEvent(
-			types.EventTypePurchaseShield,
-			sdk.NewAttribute(types.AttributeKeyPoolID, strconv.FormatUint(msg.PoolID, 10)),
-			sdk.NewAttribute(types.AttributeKeyShield, msg.Shield.String()),
+			types.EventTypeDepositCollateral,
+			sdk.NewAttribute(types.AttributeKeyCollateral, msg.Collateral.String()),
 			sdk.NewAttribute(sdk.AttributeKeySender, msg.From.String()),
 		),
 	})
 	return &sdk.Result{Events: ctx.EventManager().Events()}, nil
 }
 
-func handleMsgWithdrawRewards(ctx sdk.Context, msg types.MsgWithdrawRewards, k Keeper) (*sdk.Result, error) {
-	amount, err := k.PayoutNativeRewards(ctx, msg.From)
-	if err != nil {
-		return &sdk.Result{Events: ctx.EventManager().Events()}, err
+func handleMsgWithdrawCollateral(ctx sdk.Context, msg types.MsgWithdrawCollateral, k Keeper) (*sdk.Result, error) {
+	if msg.Collateral.Denom != k.BondDenom(ctx) {
+		return nil, types.ErrCollateralBadDenom
+	}
+
+	if err := k.WithdrawCollateral(ctx, msg.From, msg.Collateral.Amount); err != nil {
+		return nil, err
 	}
+
 	ctx.EventManager().EmitEvents(sdk.Events{
 		sdk.NewEvent(
-			types.EventTypeWithdrawRewards,
-			sdk.NewAttribute(types.AttributeKeyAccountAddress, msg.From.String()),
-			sdk.NewAttribute(types.AttributeKeyDenom, k.BondDenom(ctx)),
-			sdk.NewAttribute(types.AttributeKeyAmount, amount.String()),
-		),
-		sdk.NewEvent(
-			sdk.EventTypeMessage,
-			sdk.NewAttribute(sdk.AttributeKeyModule, types.AttributeValueCategory),
+			types.EventTypeWithdrawCollateral,
+			sdk.NewAttribute(types.AttributeKeyCollateral, msg.Collateral.String()),
```

Comment:
what's up with this change?

Reply:
Event for withdrawing collateral is not changed.